timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -64,6 +79,29 @@ public void bringAllNodesOnline() {
     }
 
     @Test
+    public void blockedLockRequestThrows503OnLeaderElection() throws InterruptedException {
+        LockRefreshToken lock = CLUSTER.lock(CLIENT_1, BLOCKING_LOCK_REQUEST);
+        assertThat(lock).isNotNull();
+
+        TestableTimelockServer leader = CLUSTER.currentLeader();
+        Future<LockRefreshToken> lock2 = Executors.newSingleThreadExecutor()
+                .submit(() -> leader.lock(CLIENT_2, BLOCKING_LOCK_REQUEST));
+
+        Uninterruptibles.sleepUninterruptibly(5, TimeUnit.SECONDS);
+        CLUSTER.nonLeaders().forEach(TestableTimelockServer::kill);
+        // Lock on leader so that AwaitingLeadershipProxy notices leadership loss.
+        assertThatThrownBy(() -> leader.lock(CLIENT_3, BLOCKING_LOCK_REQUEST))
+                .satisfies(ExceptionMatchers::isRetryableExceptionWhereLeaderCannotBeFound);
+        try {
+            lock2.get();
+            fail(String.format("Lock client %s should have thrown an execution exception as leader is not available",
+                    CLIENT_2));
+        } catch (ExecutionException e) {
```

Comment:
nit: use `assertThatThrownBy(() -> lock2.get()).getCause().satisfies(...)` (not sure if that exact syntax works)